mongodbatlas/resource_mongodbatlas_org_invitation.go

```diff
@@ -0,0 +1,269 @@
+		Schema: map[string]*schema.Schema{
+			"org_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"invitation_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"roles": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasOrgInvitationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	ids := decodeStateID(d.Id())
+	orgID := ids["org_id"]
+	username := ids["username"]
+	invitationID := ids["invitation_id"]
+
+	orgInvitation, resp, err := conn.Organizations.Invitation(context.Background(), orgID, invitationID)
+	if err != nil {
+		// case 404
+		// deleted in the backend case
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf("error getting Organization Invitation information: %s", err))
+	}
+
+	if err := d.Set("username", orgInvitation.Username); err != nil {
+		return diag.FromErr(fmt.Errorf("error getting `username` for Organization Invitation (%s): %s", d.Id(), err))
+	}
+
+	if err := d.Set("org_id", orgInvitation.GroupID); err != nil {
+		return diag.FromErr(fmt.Errorf("error getting `username` for Organization Invitation (%s): %s", d.Id(), err))
+	}
+
+	if err := d.Set("invitation_id", orgInvitation.ID); err != nil {
+		return diag.FromErr(fmt.Errorf("error getting `invitation_id` for Organization Invitation (%s): %s", d.Id(), err))
+	}
+
+	if err := d.Set("expires_at", orgInvitation.ExpiresAt); err != nil {
+		return diag.FromErr(fmt.Errorf("error getting `expires_at` for Organization Invitation (%s): %s", d.Id(), err))
+	}
+
+	if err := d.Set("created_at", orgInvitation.CreatedAt); err != nil {
+		return diag.FromErr(fmt.Errorf("error getting `created_at` for Organization Invitation (%s): %s", d.Id(), err))
+	}
+
+	if err := d.Set("roles", orgInvitation.Roles); err != nil {
+		return diag.FromErr(fmt.Errorf("error getting `roles` for Organization Invitation (%s): %s", d.Id(), err))
+	}
+
+	d.SetId(encodeStateID(map[string]string{
+		"username":      username,
+		"org_id":        orgID,
+		"invitation_id": invitationID,
+	}))
+
+	return nil
+}
+
+func resourceMongoDBAtlasOrgInvitationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	orgID := d.Get("org_id").(string)
+
+	err := validateOrgRoles(d.Get("roles").(*schema.Set))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	InvitationReq := &matlas.Invitation{
```

Comment:
[style] starting with an uppercase has special meaning so let's not use it if not needed 
```suggestion
	invitationReq := &matlas.Invitation{
```

Reply:
Fixed